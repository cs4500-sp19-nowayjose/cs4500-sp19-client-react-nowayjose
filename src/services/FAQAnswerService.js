export default class FAQAnswerService {
    baseurl = "https://cs4500-sp19-nowayjose.herokuapp.com"
    static instance = null;
    static getInstance() {
        if(FAQAnswerService.instance === null) {
            FAQAnswerService.instance = new FAQAnswerService()
        }
        return this.instance
    }

    findFAQAnswerById = id =>
        fetch(this.baseurl + `/api/faq-answers/${id}`)
            .then(response => response.json())

    findAllFAQAnswers = () =>
        fetch(this.baseurl + "/api/faq-answers")
            .then(response => response.json())
    
    createFAQAnswer(faqAnswer) {
        return fetch(this.baseurl + "/api/faq-answers", {
            method: 'post',
            headers: {
                'content-type': 'application/json'
            },
            body: JSON.stringify(faqAnswer)
        })
        .then(response => response.json())
    }

    deleteFAQAnswer(faqAnswer) {
        return fetch(this.baseurl + "/api/faq-answers/" + faqAnswer.id, {
            method: 'delete',
            headers: {
                'content-type': 'application/json'
            }
        })
    }

    updateFAQAnswer(faqAnswer) {
        return fetch(this.baseurl + "/api/faq-answers/" + faqAnswer.id, {
            method: 'put',
            body: JSON.stringify(faqAnswer),
            headers: {
                'content-type': 'application/json'
            },
        })
        .then(function (response) {
            return response.json()
        })
    }
}