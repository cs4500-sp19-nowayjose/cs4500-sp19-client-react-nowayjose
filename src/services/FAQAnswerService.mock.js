import frequentlyAskedAnswers from '../data/frequentlyAskedAnswers.mock.json'

function setUp() {
    global.fetch = jest.fn()
        .mockImplementation((url, config) => {
            if (!config) {
                return handleGet(url);
            }
            const {method, body} = config;
            switch(method.toUpperCase()) {
                case "DELETE":
                    return handleDelete(url)
                case "POST":
                    return handlePost(url, body)
                default:
                    return {"default": "default"}
            }
        });
}

function handleGet(url) {
    const returnData = {};
    if (url.search(/\/api\/faq-answers\/d+/) !== -1) {
        const aid = parseInt(url.slice(url.search(/\/d+\b/) + 1))
        returnData.json = function() {
            return frequentlyAskedAnswers.filter(({id}) => id === aid)[0]
        }
    }
    else if (url.includes("/faq-answers")) {
        returnData.json = function() {
            return frequentlyAskedAnswers
        }
    }
    return new Promise(resolve => resolve(returnData))
}

function handleDelete(url) {
    return new Promise(resolve => resolve());
}

function handlePost(url, body) {
    let ans = JSON.parse(body);
    ans.id = (new Date()).getTime();
    frequentlyAskedAnswers.push(ans)
    return new Promise((resolve, reject) => {
        resolve({json: function() {
            return frequentlyAskedAnswers
        }})
    })
}

export default setUp