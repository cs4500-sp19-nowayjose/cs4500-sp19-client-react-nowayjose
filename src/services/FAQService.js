export default class FAQService {
    static instance = null;
    static getInstance() {
        if(FAQService.instance === null) {
            FAQService.instance = new FAQService()
        }
        return this.instance
    }
   
    findFAQById = id =>
        fetch(`https://cs4500-sp19-nowayjose.herokuapp.com/api/faqs/${id}`)
            .then(response => response.json())
    findAllFAQs = () =>
        fetch("https://cs4500-sp19-nowayjose.herokuapp.com/api/faqs")
            .then(response => response.json())
}