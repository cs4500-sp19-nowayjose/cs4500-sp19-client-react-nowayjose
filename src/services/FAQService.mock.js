global.fetch = jest.fn()
    .mockImplementation(url => {
        if (url.includes("/faqs/3")) {
            return new Promise((resolve, reject) => {
                resolve({
                    json: function () {
                        return { "id": 3, "title": "charlie", "question": "c" }
                    }
                });
            });
        }
        else if (url.includes("/faqs")) {
            return new Promise((resolve, reject) => {
                resolve({
                    json: function () {
                        return [
                            { "id": 1, "title": "alice", "question": "a" },
                            { "id": 2, "title": "bob", "question": "b" },
                            { "id": 3, "title": "charlie", "question": "c" },]
                    }
                });
            });
        }

        return { "default": "default" }
    });