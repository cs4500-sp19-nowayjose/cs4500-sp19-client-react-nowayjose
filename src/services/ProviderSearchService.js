export default class ProviderSearchService {
  static instance = null;
  static host = 'https://cs4500-sp19-nowayjose.herokuapp.com'
  static getInstance() {
      if(ProviderSearchService.instance === null) {
          ProviderSearchService.instance = new ProviderSearchService();
      }
      return this.instance;
  }
  findAllProvidersForServiceId = id =>
    fetch(`${ProviderSearchService.host}/api/provider-search/${id}`)
      .then(response => response.json());
  
  searchProviders = (zip, title) => 
    fetch(`${ProviderSearchService.host}/api/service-provider/filter?title=${title}&zip=${zip}`)
      .then(response => response.json())
}
