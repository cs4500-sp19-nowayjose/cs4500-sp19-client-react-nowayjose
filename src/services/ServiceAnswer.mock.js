import serviceAnswerMockData from '../data/serviceAnswers.mock.json'

function setUp() {
  global.fetch = jest.fn()
    .mockImplementation((url, config) => {
      if (!config) {
        return handleGet(url);
      }
      const { method, body } = config;
      switch (method.toUpperCase()) {
        case "DELETE":
          return handleDelete(url)
        default:
          return { "default": "default" }
      }
    });
}

function handleGet(url) {
  const returnData = {};
  // mock findServiceAnswerById
  if (url.search(/\/api\/service_question_answers\/\d+/) !== -1) {
    const questionId = parseInt(url.slice(url.search(/\/\d+\b/) + 1));
    returnData.json = function() {
      return serviceAnswerMockData.filter(({ id }) => id === questionId)[0];
    }
  // mock findAllServiceAnswers
  } else if (url.indexOf('api/service_question_answers') !== -1) {
    returnData.json = function() {
      return serviceAnswerMockData
    }
  }
  return new Promise(resolve => resolve(returnData))
}

function handleDelete(url) {
  // delete is a void function
  return new Promise(resolve => resolve());
}

export default setUp;
