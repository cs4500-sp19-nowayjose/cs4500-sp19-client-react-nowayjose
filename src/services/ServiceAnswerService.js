export default class ServiceAnswerService {
    static instance = null
    static url = 'https://cs4500-sp19-nowayjose.herokuapp.com/'
    static url = 'http://localhost:8080/'
    static getInstance() {
        if(ServiceAnswerService.instance === null) {
            ServiceAnswerService.instance = new ServiceAnswerService()
        }
        return this.instance
    }
    findServiceAnswerById = id =>
        fetch(`${ServiceAnswerService.url}api/service_question_answers/${id}`)
            .then(response => response.json())
    findAllServiceAnswers = () =>
        fetch(`${ServiceAnswerService.url}api/service_question_answers`)
            .then(response => response.json())
    delete = id =>
        fetch(`${ServiceAnswerService.url}api/service_question_answers/${id}`, {method: "DELETE"})
    create = body =>
        fetch(`${ServiceAnswerService.url}api/service_question_answers`,
          {
            method: "POST",
            headers: {
              'Content-Type': 'application/json'
            },
            body: JSON.stringify(body)
          })
}
