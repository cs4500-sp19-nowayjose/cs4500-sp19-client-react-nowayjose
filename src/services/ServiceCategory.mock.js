import serviceCategoryMockData from '../data/service-categories.mock.json'

function setUp() {
  global.fetch = jest.fn()
    .mockImplementation((url, config) => {
      if (!config) {
        return handleGet(url);
      }
      const { method, body } = config;
      switch (method.toUpperCase()) {
        case "DELETE":
          return handleDelete(url)
        case "POST":
          return handlePost(url, body)
        default:
          return { "default": "default" }
      }
    });
}

function handleGet(url) {
  const returnData = {};
  // mock findAllServiceForCategory
  if (url.search(/\/api\/categories\/\d+/) !== -1) {
    const categoryId = parseInt(url.slice(url.search(/\/\d+\b/) + 1));
    returnData.json = function() {
      return serviceCategoryMockData.filter(({ id }) => id === categoryId)[0]
    }
  // mock findAllServiceCategories
  } else if (url.indexOf('api/categories') !== -1) {
    returnData.json = function() {
      return serviceCategoryMockData
    }
  }
  return new Promise(resolve => resolve(returnData))
}

function handleDelete(url) {
  // delete is a void function
  return new Promise(resolve => resolve());
}

function handlePost(url, body) {
  const returnData = {};
//   const { title, description } = JSON.parse(body);
//   // mock findServiceQuestionByCriteria
//   if (url.indexOf('api/service_question/filter') !== -1) {
//     returnData.json = function() {
//       return serviceQuestionMockData.filter(q => {
//         return q.title.indexOf(title || '') !== -1 && q.description.indexOf(description || '') !== -1
//       })
//     }
//   }
  return new Promise(resolve => resolve(returnData));
}

export default setUp;