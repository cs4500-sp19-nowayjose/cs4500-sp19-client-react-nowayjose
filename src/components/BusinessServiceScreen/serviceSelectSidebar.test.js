import React from 'react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { act } from 'react-dom/test-utils'

import ServiceSelectSidebar from './serviceSelectSidebar';
import services from '../../data/services.mock.json'
import setUp from '../../services/ServiceService.mock.js'

beforeAll(function () {
    setUp();
})

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
})

afterEach(() => {
  document.body.removeChild(container);
  container = null;
})

test('it selects a service', function () {
    var called = false;
    act(() => {
      ReactDOM.render(
        <ServiceSelectSidebar
          query="a"
          setQuery={() => {}}
          possibleServices={services}
          setPossibleServices={() => {}}
          activeServiceId={0}
          addService={() => {}}
          removeService={() => {}}
          selectService={() => {called = true;}}
          selectedServices={services}
          />
      , container)
    });
    document.body.querySelector(".select-service-button").dispatchEvent(new MouseEvent('click', {bubbles: true}));
    expect(called).toBe(true);
})

test('it renders properly', function() {
  const tree = renderer
    .create(
        <ServiceSelectSidebar
          query="a"
          setQuery={() => {}}
          possibleServices={services}
          setPossibleServices={() => {}}
          activeServiceId={0}
          addService={() => {}}
          removeService={() => {}}
          selectService={() => {called = true;}}
          selectedServices={services}
          />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
})
