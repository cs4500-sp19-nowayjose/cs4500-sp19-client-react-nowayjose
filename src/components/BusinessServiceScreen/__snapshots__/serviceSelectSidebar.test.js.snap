// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders properly 1`] = `
<div
  className="service-search"
>
  <h4>
    Search Services
  </h4>
  <input
    onChange={[Function]}
    placeholder="Search services"
    style={
      Object {
        "width": "100%",
      }
    }
    type="text"
    value="a"
  />
  <div
    className="service-search-results"
  >
    <button
      className="btn btn-light"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
          "margin": "5px auto",
          "width": "90%",
        }
      }
    >
      The Service Squad
    </button>
    <button
      className="btn btn-light"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
          "margin": "5px auto",
          "width": "90%",
        }
      }
    >
      The Second Service
    </button>
    <button
      className="btn btn-light"
      onClick={[Function]}
      style={
        Object {
          "display": "block",
          "margin": "5px auto",
          "width": "90%",
        }
      }
    >
      Services R Us
    </button>
  </div>
  <div
    className="selected-services"
  >
    <h4>
      Selected Services
    </h4>
    <button
      className="btn btn-light"
      style={
        Object {
          "display": "block",
          "margin": "5px auto",
          "width": "90%",
        }
      }
    >
      <span
        className="select-service-button"
        onClick={[Function]}
      >
        The Service Squad
      </span>
      <span
        onClick={[Function]}
      >
         x
      </span>
    </button>
    <button
      className="btn btn-light"
      style={
        Object {
          "display": "block",
          "margin": "5px auto",
          "width": "90%",
        }
      }
    >
      <span
        className="select-service-button"
        onClick={[Function]}
      >
        The Second Service
      </span>
      <span
        onClick={[Function]}
      >
         x
      </span>
    </button>
    <button
      className="btn btn-light"
      style={
        Object {
          "display": "block",
          "margin": "5px auto",
          "width": "90%",
        }
      }
    >
      <span
        className="select-service-button"
        onClick={[Function]}
      >
        Services R Us
      </span>
      <span
        onClick={[Function]}
      >
         x
      </span>
    </button>
  </div>
</div>
`;
